synapse/storage/database.py

```diff
@@ -1483,6 +1486,9 @@ def simple_upsert_many_txn(
             value_values: A list of each row's value column values.
                 Ignored if value_names is empty.
         """
+        if not value_values:
```

Comment:
I think this needs to be `not key_values and not value_values`

Reply:
Hah, I just changed that. 👍

Reply:
haha, great minds